runtime/runtime/src/ext.rs

```diff
@@ -228,30 +228,30 @@ impl<'a> External for RuntimeExt<'a> {
         &mut self,
         data_id: CryptoHash,
         data: Vec<u8>,
-    ) -> Result<(), VMLogicError> {
+    ) -> Result<bool, VMLogicError> {
         // If the yielded promise was created by a previous transaction, we'll find it in the trie
         if let Some(yielded_promise) =
             get_yielded_promise(self.trie_update, data_id).map_err(wrap_storage_error)?
         {
             // Yields are only resumable by the account which created them
-            if yielded_promise.account_id != *self.account_id {
-                return Err(
-                    HostError::YieldedPromiseNotFound { data_id: data_id.to_string() }.into()
```

Comment:
Is `HostError::YieldedPromiseNotFound` used anywhere after this change? Can we get rid of it?